tensorflow/python/kernel_tests/segment_reduction_ops_test.py

```diff
@@ -641,6 +646,56 @@ def testGradientSegmentsInvalid4(self):
         with self.assertRaisesOpError(r"Segment id 0 out of range \[0, 0\)"):
           s.eval()
 
+def type_to_str(t):
+  if t == np.float32:
+    return "fp32"
+  if t == np.float64:
+    return "fp64"
+
+class SegmentReductionOpBenchmark(test.Benchmark):
+
+  def benchmarkSegmentSumGPU(self):
+    repeat = 10
+    outer_dim_options = [2**x for x in range(7, 14, 2)]
+    ratio_options = [2**x for x in range(1, 6, 2)]
+    inner_dim_options = [2**x for x in range(7, 14, 2)]
+    dtype_options = [np.float32, np.float64]
+
+    for outer_dim, ratio, inner_dim, dtype in \
+      itertools.product(outer_dim_options,
+                        ratio_options, inner_dim_options, dtype_options):
+      output_outer_dim = int(outer_dim/ratio)
+
+      const = np.random.randint(5, size=(outer_dim, inner_dim))
+      seg_ids = np.sort(np.random.randint(
+          output_outer_dim, size=outer_dim))
+
+      op_functors = [lambda vc, vs:
```

Comment:
Most benchmarks would split them into two different sub-benchmarks. They can share the underlying implementation